src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden !== true) {
+      target.health -= 50;
+
+      if (target.health === 0) {
+        Animal.alive = Animal.alive.filter(animal => animal !== target);
+      }
+    }
```

Comment:
Try to avoid nested "if" statements.
Also, `target.hidden !== true` you simplify `!target.hidden`